contrib/endpoints/src/api_manager/config_manager.cc

```diff
@@ -60,43 +70,20 @@ void ConfigManager::Init(RolloutApplyFunction rollout_apply_function) {
 
 void ConfigManager::OnFetchMetadataDone(utils::Status status) {
   if (!status.ok()) {
-    // We should not get here
     global_context_->env()->LogError("Unexpected status: " + status.ToString());
-    rollout_apply_function_(utils::Status(Code::ABORTED, status.ToString()),
-                            kEmptyConfigs);
     return;
   }
 
-  // Update service_name
   if (global_context_->service_name().empty()) {
     global_context_->set_service_name(
         global_context_->gce_metadata()->endpoints_service_name());
   }
 
-  // Update config_id
-  if (global_context_->config_id().empty()) {
-    global_context_->config_id(
-        global_context_->gce_metadata()->endpoints_service_config_id());
-  }
-
-  // Call ApiManager with status Code::ABORTED, ESP will stop moving forward
   if (global_context_->service_name().empty()) {
-    std::string msg = "API service name is not specified";
-    global_context_->env()->LogError(msg);
-    rollout_apply_function_(utils::Status(Code::ABORTED, msg), kEmptyConfigs);
+    global_context_->env()->LogError("API service name is not specified");
     return;
   }
 
-  // TODO(jaebong) config_id should not be empty for the first version
-  // This part will be removed after the rollouts feature added
-  if (global_context_->config_id().empty()) {
-    std::string msg = "API config_id is not specified";
-    global_context_->env()->LogError(msg);
-    rollout_apply_function_(utils::Status(Code::ABORTED, msg), kEmptyConfigs);
-    return;
-  }
-
-  // Fetch service account token
   GlobalFetchServiceAccountToken(global_context_, [this](utils::Status status) {
     OnFetchAuthTokenDone(status);
```

Comment:
no need to call metadata to fetch token at Init().   The actual "remote call" needs to fetch token at the time of call.  token is time sensitive.